skyportal/handlers/api/annotation.py

```diff
@@ -30,18 +49,64 @@ def get(self, annotation_id):
               application/json:
                 schema: Error
         """
-        annotation = Annotation.get_if_accessible_by(
-            annotation_id, self.current_user, raise_if_none=True
-        )
+
+        try:
+            annotation_id = int(annotation_id)
+        except (TypeError, ValueError):
+            return self.error("Must provide a valid annotation ID. ")
+        # the default is to annotate an object
+        if associated_resource_type.lower() in (
+            "object",
+            "objects",
+            "source",
+            "sources",
+        ):
+            annotation = Annotation.get_if_accessible_by(
+                annotation_id, self.current_user, raise_if_none=True
```

Comment:
None of the `raise_if_none` exceptions seem to be handled.

Reply:
I think you are reviewing a previous commit. Let me tag you again after I push a couple more changes.